modules/core/src/main/java/org/apache/ignite/internal/processors/datastructures/GridCacheCountDownLatchImpl.java

```diff
@@ -236,6 +236,9 @@ public GridCacheCountDownLatchImpl(String name,
     @Override public void onUpdate(int cnt) {
         assert cnt >= 0;
 
+        if (initGuard.get())
```

Comment:
Vlad, lets use the following implementation for "onUpdate" method

------
        CountDownLatch latch0 = internalLatch;

        if (initGuard.get() && latch0 == null) {
            U.awaitQuiet(initLatch);

            latch0 = internalLatch;
        }

        assert latch0 != null;

        while (latch0.getCount() > cnt)
            latch0.countDown();
------

This implementation:
- avoids unnecessary calls to  U.awaitQuiet(initLatch); when the latch is already initialized
- avoids frequent reads from volatile variable